dali/operators/decoder/video/video_decoder_base.h

```diff
@@ -20,8 +20,8 @@
 #include "dali/operators/reader/loader/video/frames_decoder.h"
 #include "dali/operators/reader/loader/video/frames_decoder_gpu.h"
```

Comment:
Maybe it would make sense to move these dependencies to the files with particular decoders? Especially the GPU part. Right now, we can build with FFMPEG=on and NVDEC=off, so only the CPU variant of `VideoReaderDecoder`. When this base class has both of the frame decoders it will not be possible to build CPU variant without the GPU one. 
I am not saying that this is super important feature, but since we have that, probably it make sense to keep it.

Reply:
Done, by making VideoDecoderBase generic over the FramesDecoder type